.github/workflows/publish.yml

```diff
@@ -0,0 +1,88 @@
+name: Update Version, Test, Publish to PyPI, and Upload Release Artifacts
+
+on:
+  release:
+    types: [created, edited]
+
+jobs:
+  build-test-publish-upload:
+    runs-on: ubuntu-latest
+    environment: release
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install hatch
+    - name: Set release version
+      run: |
+        # Get the tag from the GitHub release
+        TAG=${GITHUB_REF#refs/tags/}
+        # Remove 'v' prefix if present
+        VERSION=${TAG#v}
+        hatch version $VERSION
+    - name: Build package
+      run: hatch build
+    - name: Run tests
+      run: hatch run test:pytest
+    - name: Retrieve and mint OIDC token
+    # (https://docs.pypi.org/trusted-publishers/using-a-publisher/)
+      run: |
```

Comment:
It is much simpler with this setup: https://github.com/projectmesa/mesa/blob/main/.github/workflows/release.yml. It uses `pypa/gh-action-pypi-publish@release/v1`.

Reply:
thanks! I'll update that